### PR TITLE
Dr #65 sad path testing

### PR DESCRIPTION
**Pull request by drod1000**
@case-eee  @Dpalazzari @jdconrad89  @kheppenstall @Robbie-Smith 
Added sad paths for unsuccessful account creation, login, and failing to update a project

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -4,21 +4,24 @@ def new
   end
 
   def create
+    @user = User.find_by(user_params) if params[:user]
     if with_facebook?
       @user = User.find_or_create_by_omniauth(oauth_user_data)
       successful_login
-    elsif !with_facebook?
-      @user = User.find_by(user_params)
+    elsif @user && !with_facebook?
```
Comment (Robbie-Smith) [inline app/controllers/sessions_controller.rb]: @drod1000 why the change here?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -11,17 +11,22 @@ def create
...
 
   def show
     @user = User.find(params[:id])
   end
 
+  def destroy
+    user = User.find(params[:id])
+    user.take_offline
+    redirect_to users_path, success: "#{user.name} taken offline"
+  end
```
Comment (Robbie-Smith) [inline app/controllers/users_controller.rb]: @drod1000 Why not use the destroy action in the sessions controller? I think that the destroy action on the users controller would only be used for deleting a user from the database, and the sessions controller should handle the taking someone offline. Thoughts?

File: spec/features/project_owner/project_owner_edits_project_spec.rb [hunk 3]
```diff
@@ -5,22 +5,33 @@
...
 
   context "visits their own project" do
-    it "and edits their project" do
+    before do
       make_project_owner(owner, project)
       visit project_path(project.slug)
 
       expect(page).to have_content("Bob's bean maker")
       expect(page).to have_content("Howdy ho")
       click_on("Edit Project")
```
Comment (Robbie-Smith) [inline spec/features/project_owner/project_owner_edits_project_spec.rb]: expectations shouldn't live in a before block

Comment (Robbie-Smith) [inline spec/features/project_owner/project_owner_edits_project_spec.rb]: @drod1000 expectations shouldn't live in a before block

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (Dpalazzari) [inline app/controllers/sessions_controller.rb]: I am curious about this as well.

File: spec/support/feature_helpers.rb [hunk 4]
```diff
@@ -12,9 +12,39 @@ def logged_in_admin
     admin
   end
```
Comment (Robbie-Smith) [inline spec/support/feature_helpers.rb]: @drod1000 @kheppenstall  Why do we have so many stubbed login methods?

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (drod1000) [inline app/controllers/sessions_controller.rb]: @Robbie-Smith the way it was set up before it wasn't checking to see whether the user was valid before going into that block. If they tried to sign in without Facebook but provided invalid credentials that didn't allow us to find a user the successful login method would to set the session to @user.id when it's nil and we'd get an error.

Comment (Robbie-Smith) [inline app/controllers/sessions_controller.rb]: @drod1000 Isn't that what we wanted? If someone provides invalid credentials, to be redirected back to the login screen.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Dpalazzari) [inline app/controllers/users_controller.rb]: Are we taking them offline as in permanently deleting the user, or are we just destroying their session?

Comment (Robbie-Smith) [inline app/controllers/users_controller.rb]: @Dpalazzari Thanks, for clarifying my question.

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (drod1000) [inline app/controllers/sessions_controller.rb]: @Robbie-Smith That's not the block it was hitting though. It was hitting the successful login block which resulted in a undefined method `id' for nil:NilClass for the user

Comment (Robbie-Smith) [inline app/controllers/sessions_controller.rb]: @drod1000 Oh okay. Thanks for clarifying.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (drod1000) [inline app/controllers/users_controller.rb]: @Robbie-Smith @Dpalazzari I believe this is part of Kyle's PR.

Comment (Robbie-Smith) [inline app/controllers/users_controller.rb]: @kheppenstall
